### PR TITLE
Add implementation for GET /sports/ route; and fix routing of auth module

### PR DESCRIPTION
**Pull request by GToidZ**
## Overview
This PR adds and modifies the system by:
* Added a working implementation for GET /sports/ route.
* Fixed the name for auth keygen route.
* Fixed the routing hierarchy for auth module.
* Added examples for environment variables configuration.

## Further Improvement
We should implement a cache system, so that pymongo wouldn't need to always aggregate data.

### CODE REVIEW

File: routers/sports_router.py [hunk 1]
```diff
@@ -1,12 +1,29 @@
...
+                "$replaceRoot": {
+                    "newRoot": {
+                        "$arrayToObject": [
+                            [{"k": {"$toString": "$sport_id"}, "v": "$sport_name"}]
+                        ]
+                    }
+                }
+            }
+        ]
+    )
```
Comment (GToidZ) [inline routers/sports_router.py:22]: ## How It Works
We aggregate from `SportDetail` collection that we only want array of key-value pairs where `sport_id` as key and `sport_name` as value.

`$replaceRoot` allows us to customize how the J/BSON output should be, by stating the root to start from inner nodes or entirely new nodes. See links below to learn more:

* https://www.mongodb.com/docs/manual/reference/operator/aggregation/replaceRoot/#-replaceroot-with-a-newly-created-document
* https://www.mongodb.com/docs/manual/reference/operator/aggregation/replaceRoot/#-replaceroot-with-an-embedded-document-field

**Review by Supakrit65 — APPROVED** [review]
Doing **Great**

**Review by Halcyon905 — APPROVED** [review]
click me
nice

**Review by Savetang19 — APPROVED** [review]
LGTM - approve ka
LGTM

**Pull request merged**